### PR TITLE
docs: adds note about branches and tags in contributing

### PR DESCRIPTION
**Pull request by stockholmux**
Signed-off-by: Kyle J. Davis <kyledvs@amazon.com>

**Issue number:** n/a

**Description of changes:** Documents how tags and branches work in contributing markdown.

**Testing done:** Eyes used.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by webern — APPROVED** [review]

**Review by mchaker — APPROVED** [review]
🚀

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,6 +39,11 @@ To send us a pull request, please:
 
 GitHub provides additional documentation on [forking a repository](https://help.github.com/articles/fork-a-repo/) and [creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
 
+## Repo branch and tag structure 
+
+Active development occurs under the `develop` branch.
+
+Bottlerocket uses both tags and branches for release alignment. Numbered releases are always associated with [tags that mirror the full SemVer 3-digit version number](https://github.com/bottlerocket-os/bottlerocket/tags) (e.g. `1.7.2`). [Branches are for patching only](https://github.com/bottlerocket-os/bottlerocket/branches/all): if a patch is required, a branch will be cut for that minor release line (e.g. `1.7.x`). As a consequence, some previous minor versions may not have a branch if they never required a subsquent patch.
```
Comment (bcressey) [inline CONTRIBUTING.md]:
```suggestion
Bottlerocket uses both tags and branches for release alignment. Numbered releases are always associated with [tags that mirror the full SemVer 3-digit version number](https://github.com/bottlerocket-os/bottlerocket/tags) (e.g. `1.7.2`). [Branches are for patching only](https://github.com/bottlerocket-os/bottlerocket/branches/all): if a patch is required, a branch will be cut for that minor release line (e.g. `1.7.x`). As a consequence, some previous minor versions may not have a branch if they never required a subsequent patch.
```

**Review by bcressey — APPROVED** [review]
LGTM apart from the typo

File: CONTRIBUTING.md (hunk 1 above)
Comment (stockholmux) [reply CONTRIBUTING.md]: Fixed, squashed.

**Pull request merged**